*** source/gdb-7.4/gdb/python/python.c	2012-02-01 10:39:59.812953656 +0100
--- staging/gdb-7.4/gdb/python/python.c	2012-02-01 10:33:45.842953480 +0100
***************
*** 32,38 ****
  #include "serial.h"
  #include "readline/tilde.h"
  #include "python.h"
  #include "cli/cli-utils.h"
!
  #include <ctype.h>
  
--- 32,39 ----
  #include "serial.h"
  #include "readline/tilde.h"
  #include "python.h"
  #include "cli/cli-utils.h"
! #include <libgen.h>
+ #include <mach-o/dyld.h>
  #include <ctype.h>
  
***************
*** 1273,1278 ****
--- 1274,1293---
  			     SLASH_STRING, "python", NULL));
  #endif
  
+ 
+   int bufsize = 1024;
+   char readlinkbuffer[bufsize];
+   int readlinks = _NSGetExecutablePath(readlinkbuffer, &bufsize);
+   if (readlinks == -1) {//readlinkbuffer[readlinks] = 0; 
+       
+       return;
+   }
+   char *executeablepath = dirname(readlinkbuffer);
+   char *pythonhome = malloc(strlen(executeablepath) + strlen("/python/") + 2);
+   strcpy(pythonhome, executeablepath);
+   strcat(pythonhome, "/python/");
+   setenv("PYTHONHOME", pythonhome, 1);
+ 
    Py_Initialize ();
    PyEval_InitThreads ();
  
